### ISSUE TITLE
Notes-Table: `permission` is an enum and it misses values

### ISSUE THREAD

**Issue by ccoenen** [hackmdio/hackmd#752]
I can't switch to protected or limited permissions. The setting will get accepted and is in place as long as someone has a pad open, but it is never committed to database, so the next time I open the same pad, the value is _empty_!

Here's why: My `Notes` table looks like this

```
MariaDB [hackmd]> show fields from Notes;
+------------------+----------------------------------------------+------+-----+---------+-------+
| Field            | Type                                         | Null | Key | Default | Extra |
+------------------+----------------------------------------------+------+-----+---------+-------+
| id               | char(36)                                     | NO   | PRI |         |       |
| shortid          | varchar(255)                                 | NO   | UNI | NULL    |       |
| alias            | varchar(255)                                 | YES  | UNI | NULL    |       |
| permission       | enum('freely','editable','locked','private') | YES  |     | NULL    |       |
| viewcount        | int(11)                                      | NO   |     | 0       |       |
| title            | text                                         | YES  |     | NULL    |       |
| content          | longtext                                     | YES  |     | NULL    |       |
| authorship       | text                                         | YES  |     | NULL    |       |
| lastchangeAt     | datetime                                     | YES  |     | NULL    |       |
| savedAt          | datetime                                     | YES  |     | NULL    |       |
| createdAt        | datetime                                     | NO   |     | NULL    |       |
| updatedAt        | datetime                                     | NO   |     | NULL    |       |
| deletedAt        | datetime                                     | YES  |     | NULL    |       |
| ownerId          | char(36)                                     | YES  |     | NULL    |       |
| lastchangeuserId | char(36)                                     | YES  |     | NULL    |       |
+------------------+----------------------------------------------+------+-----+---------+-------+
```

the type of `permission` seems wrong: the enum is missing "limited" and "protected". Can anyone else confirm this, or did I somehow break my installation?

Should this field be an enum in the first place? (the [model](https://github.com/hackmdio/hackmd/blob/1.0.1-ce/lib/models/note.js#L44-L47) suggests, that it _is_ an enum).

Would there have to be a migration for that? From the currently existing migrations I would have guessed this should be a string instead (only migration to touch it appears to be [update-to-0-3-1](https://github.com/hackmdio/hackmd/blob/master/lib/migrations/20150702001020-update-to-0_3_1.js#L13-L17), which I probably never ran as it has been added _after_ I installed my setup long before those "old" migrations were added. (and they obviously gave me problems while upgrading to 1.0.1-ce, as the tables and fields were already present). The feature seems to have been introduced [here](https://github.com/hackmdio/hackmd/commit/89b8ddeabae81fd3a8891ce9d8191fbc9e27c83c).

**Comment by SISheogorath** [comment]
I think here is actually just a migration missing.

Because the enum is never updated. People with newer instances won't experience this issue.

So adding a new migration should fix it.

**Comment by SISheogorath** [comment]
My instance is (iirc) younger and has all permissions in the enum.

**Comment by ccoenen** [comment]
But how are they created in the first place? How does sequelize learn about those fields in case of an empty database? Does it go different routes for initial setup than for migrations?

**Comment by ccoenen** [comment]
Also, this is another place, where Things look fine while using a document, but are actually _not_ stored anywhere. We had a similar thing for #521, where only a part of the content got saved (which only surfaced once nobody was woking on a given document any longer).

There has to be some way of getting a "stored successfully" flag from the database, and in case it's not, or it produced warnings, we should burn down the editor. Before more data gets lost.

**Comment by ccoenen** [comment]
In this case, the problem might notbe "data loss", but "data leaked to people who were not supposed to see it", which i find to be equally distrurbing, depending on context.

**Comment by SISheogorath** [comment]
We actually apply some changes by HackMD itself after the migration by using: https://github.com/hackmdio/hackmd/blob/master/app.js#L232

https://sequelize.readthedocs.io/en/v3/docs/models-definition/#database-synchronization

**Comment by SISheogorath** [comment]
fixed by #757
